### PR TITLE
docs: add tips for client parallelism usage

### PR DESCRIPTION
**Pull request by ZiniuYu**
**Please merge #848 first!!!**

- [ ] Reorganize client docs
- [ ] Add tips for client parallelism

### CODE REVIEW

File: docs/user-guides/client.md [hunk 1]
```diff
@@ -1,55 +1,88 @@
...
 - Streaming: request sending is *not* blocked by the response receiving. Sending and receiving are two separate streams that run in parallel. Both are independent and each have separate internal buffer.
 - Batching: large requests are segmented into small batches and send in a stream. 
 - Low memory footprint: only load data when needed.
 - Sync/async interface: provide `async` interface that can be easily integrated into other asynchronous system.
-- Auto-detect image and sentences.
+- Auto-detect images and texts input.
 - Support gRPC, HTTP, Websocket protocols with their TLS counterparts. 
 
-This chapter introduces the API of the client. 
+This chapter introduces the usages of `clip_client`.
```
Comment (scott-martens) [inline docs/user-guides/client.md:12]: Delete this line

File: docs/user-guides/client.md [hunk 2]
```diff
@@ -1,55 +1,88 @@
...
 | `scheme` | The protocol of the server, must be one of `grpc`, `websocket`, `http`, `grpcs`, `websockets`, `https`. Protocols end with `s` are TLS encrypted. This must match with the server protocol. | `grpc`        |
 | `netloc` | The server's IP address or hostname                                                                                                                                                         | `192.168.0.3` |
 | `port`   | The public port of the server                                                                                                                                                               | `51234`       |
             
 
+
+### Connect to the CLIP server hosted by Jina AI
+
+An always-online server `api.clip.jina.ai` loaded with `ViT-L-14-336::openai` is there for you to play & test.
+Before you start, make sure you have created an access token from our [console website](https://console.clip.jina.ai/get_started),
```
Comment (scott-martens) [inline docs/user-guides/client.md:43]: ... have obtained an access token...

File: docs/user-guides/client.md [hunk 3]
```diff
@@ -1,55 +1,88 @@
...
+        'then do it better',
+        'https://picsum.photos/200',
+    ]
+)
+```
+
 ## Encoding
 
-Client provides {func}`~clip_client.client.Client.encode` function that allows you to send sentences, images to the server in a streaming and sync/async manner. Encoding here means getting the fixed-length vector representation of a sentence or image.
+`clip_client` provides {func}`~clip_client.client.Client.encode` function that allows you to send sentences, images to the server in a streaming and sync/async manner. Encoding here means getting the fixed-length vector representation of a sentence or image.
```
Comment (scott-martens) [inline docs/user-guides/client.md:75]: a _**text**_ or image.

File: docs/user-guides/client.md [hunk 4]
```diff
@@ -173,75 +217,23 @@ To get the embedding of all Documents, simply call `r.embeddings`:
 Reading an image file into bytes and put into `.blob` is possible as shown above. However, it is often unnecessary. Especially if you have a lot of images, loading all of them into memory is not a good idea. Rule of thumb, always use `.uri` and trust `clip_client` to handle it well. 
 ```
 
-### Control batch size
-
-You can specify `.encode(..., batch_size=8)` to control how many Documents is sent in each request. You can play this number and find the perfect balance between network transmission and GPU utilization.
```
Comment (scott-martens) [inline docs/user-guides/client.md:178]: ...how many Documents **_are_** sent...

Comment (scott-martens) [inline docs/user-guides/client.md:178]: You can play **_with_** this number **_to_** find...

File: docs/user-guides/client.md [hunk 5]
```diff
@@ -173,75 +217,23 @@ To get the embedding of all Documents, simply call `r.embeddings`:
 Reading an image file into bytes and put into `.blob` is possible as shown above. However, it is often unnecessary. Especially if you have a lot of images, loading all of them into memory is not a good idea. Rule of thumb, always use `.uri` and trust `clip_client` to handle it well. 
 ```
 
-### Control batch size
-
-You can specify `.encode(..., batch_size=8)` to control how many Documents is sent in each request. You can play this number and find the perfect balance between network transmission and GPU utilization. 
-
-Intuitively, setting `batch_size=1024` should give a very high GPU utilization on each request. However, a large batch size like this also means sending each request would take longer. Given that `clip-client` is designed with request and response streaming, large batch size would not benefit from the time overlapping between request streaming and response streaming.
```
Comment (scott-martens) [inline docs/user-guides/client.md:180]: ...should **_result in_** very high GPU utilization...

Comment (scott-martens) [inline docs/user-guides/client.md:180]: ...**_a_** large batch size...

File: docs/user-guides/client.md [hunk 6]
```diff
@@ -173,75 +217,23 @@ To get the embedding of all Documents, simply call `r.embeddings`:
...
-
-
-## Async encoding
-
-To encode Document in an asynchronous manner, one can use {func}`~clip_client.client.Client.aencode`.
+To encode `Document` in an asynchronous manner, one can use {func}`~clip_client.client.Client.aencode`.
 
 ```{tip}
-Despite the sexy word "async", I often found many data scientists have a misconception about asynchronous. And their motivation of using async function is often wrong. _Async is not a silver bullet._ In a simple language, you will only need `.aencode()` when there is another concurrent task that is also async. Then you want to "overlap" the time spending of these two tasks.
+Despite the sexy word "async", many data scientists may have a misconception about asynchronous. And their motivation of using async function is often wrong. _Async is not a silver bullet._ In a simple language, you will only need `.aencode()` when there is another concurrent task that is also async. Then you want to "overlap" the time spending of these two tasks.
```
Comment (scott-martens) [inline docs/user-guides/client.md:225]: ...many data scientists **_have misconceptions about asynchronous behavior_**...

File: docs/user-guides/client.md [hunk 7]
```diff
@@ -173,75 +217,23 @@ To get the embedding of all Documents, simply call `r.embeddings`:
...
 
 If your system is sync by design, there is nothing wrong about it. Go with `encode()` until you see a clear advantage of using `aencode()`, or until your boss tell you to do so.   
 ```
 
-In the following example, I have another job `another_heavylifting_job` to represent job like writing to database, downloading large file.
+In the following example, there is another job `another_heavylifting_job` to represent job like writing to database, downloading large file.
```
Comment (scott-martens) [inline docs/user-guides/client.md:230]: ... represent **_a_** job...

File: docs/user-guides/client.md [hunk 8]
```diff
@@ -458,7 +449,104 @@ c.profile('https://docarray.jina.ai/_static/favicon.png')
...
+:width: 80%
+```
+
+```{hint}
+Progress bar may not show up in the PyCharm debug terminal. This is an upstream issue of `rich` package.
+```
+
+### Processing large number of Documents
+
+Here are some suggestions when encoding large number of `Document`s:
```
Comment (scott-martens) [inline docs/user-guides/client.md:477]: ...encoding **_a_** large number...

File: docs/user-guides/client.md [hunk 9]
```diff
@@ -520,4 +608,22 @@ curl -X POST https://demo-cas.jina.ai:8443/post \
 ```text
 [-0.022064208984375,0.1044921875,...]
 [-0.0750732421875,-0.166015625,...]
+```
+
+To connect to the CLIP server hosted by Jina AI, after you have created an access token from our [console website](https://console.clip.jina.ai/get_started) or via CLI as described in [this guide](https://github.com/jina-ai/jina-hubble-sdk#create-a-new-pat), set it in the HTTP request header `Authorization` as `<your access token>`:
```
Comment (scott-martens) [inline docs/user-guides/client.md:613]: ... after you have **_received_** an access token from...

**Comment by scott-martens** [comment]
I have left a lot of comments on the client.md file. Please make a new ticket https://github.com/orgs/jina-ai/projects/34/views/1 to read and revise the entire page instead of just the changes. It need copyediting badly.

**Comment by ZiniuYu** [comment]
> I have left a lot of comments on the client.md file. Please make a new ticket https://github.com/orgs/jina-ai/projects/34/views/1 to read and revise the entire page instead of just the changes. It need copyediting badly.

@scott-martens thanks for the suggestions!! I saw that a draft ticket has already been added, so I won't add a duplicated one

**Pull request merged**